### ISSUE TITLE
[k8s 1.20] BYO Public static IP is deleted automatically after k8s service of type LoadBalancer is deleted

### ISSUE THREAD

**Issue by petermicuch** [Azure/AKS#2301]
**What happened**: After deletion of kubernetes service of type `LoadBalancer` associated **public static IP address** was also deleted.

**What you expected to happen**: When I delete kubernetes service of type `LoadBalancer` that is associated with **public static IP address** from the same or different resource group, I do not expect that the IP address would be deleted. This forces us to update DNS records with new IP if we are going to recreate the service again. This is a bug and according to description in this [article](https://docs.microsoft.com/en-us/azure/aks/static-ip), public static IP address should not be deleted when k8s service related to such IP is deleted.

**How to reproduce it (as minimally and precisely as possible)**:
* Create AKS cluster with SKU Basic LoadBalancer with contributor access to resource group with your public IPs (e.g. PUBLIC-IPS-RESOURCE-GROUP).
* Create **public static IP address** (SKU Basic) in PUBLIC-IPS-RESOURCE-GROUP resource group (e.g. 52.174.23.140).
* Create LB service that uses that **public static IP address**
```
kind: Service
apiVersion: v1
metadata:
  annotations:
    service.beta.kubernetes.io/azure-load-balancer-resource-group: PUBLIC-IPS-RESOURCE-GROUP
  name: reverse-proxy
  labels: 
    area: platform
    track: latest
spec: 
  loadBalancerIP: 52.174.23.140
  type: LoadBalancer
  selector: 
    platform-app: reverse-proxy-nginx
  ports:
  - protocol: TCP
    port: 80
    targetPort: http
    name: http
  - protocol: TCP
    port: 443
    targetPort: https
    name: https
```
* Once IP is associated with LoadBalancer, delete the service
```
kubectl delete service reverse-proxy
```
* Observe resource group PUBLIC-IPS-RESOURCE-GROUP and notice, that IP address will get deleted in few minutes.

**Environment**:
- Kubernetes version: v1.20.2
- Size of cluster: 8 nodes
- General description of workloads in the cluster: HTTP microservices
- Others: Subscription where I reproduced this multiple times - 90a3dc22-a884-4560-ab8f-45222e67333e

**Comment by abarqawi** [comment]
can you check if the allocation method is static ?
https://docs.microsoft.com/en-us/azure/virtual-network/public-ip-addresses#allocation-method

az network public-ip create     --resource-group AKSrg     --name myAKSPublicIPbasic     --sku basic     --allocation-method static

**Comment by abarqawi** [comment]
@palma21 i successfully repro the issue after upgrade cluster from 1.19.7 to 1.20.2 and it matches @petermicuch version 
we are not sure if this is new behavior in AKS 1.20.2 or not ?

note: AKS should be using basic LB

**Comment by abarqawi** [comment]
Current known issue in K8s 1.20 https://github.com/kubernetes/kubernetes/pull/100691 
do we have ETA @feiskyer @nilo19 

Mitigation: remove tags from public IP addr before deleting service.

> Tags to delete: service, kubernetes-cluster-name

> After deleting tags, controller-manager will not remove the public IP

**Comment by petermicuch** [comment]
@abarqawi - I guess removal of the tags will not cause any issues even if the service would not be deleted, right? Or would it have any influence on the LB functionality, if I remove the that tag, but keep the service running?

**Comment by ramirezalfredo** [comment]
@abarqawi I saw that behavior today with Standard LB. AKS was upgraded to 1.20.5, service was deleted with the expectation to preserve the IP, didn't happen as expected in https://docs.microsoft.com/en-us/azure/aks/static-ip, CC: @palma21

**Comment by nilo19** [comment]
It has been fixed in out-of-tree repo and cherry picked into k/k. https://github.com/kubernetes-sigs/cloud-provider-azure/issues/573

**Comment by nilo19** [comment]
@abarqawi I think it would be included in the next patch release.

**Comment by feiskyer** [comment]
The [fix](https://github.com/kubernetes/kubernetes/pull/100691) has been included in 1.20.6, and 1.21 [cherry-pick](https://github.com/kubernetes/kubernetes/pull/100693) would be merged soon.

**Comment by nilo19** [comment]
Hotfix released.
